pallets/gear/src/internal.rs

```diff
@@ -302,11 +303,21 @@ where
         }
     }
 
-    /// Charges for holding in some storage.
+    /// Charges for holding in some storage. In order to be placed into a holding storage
+    /// a message must lock some funds as a deposit to "book" the storage until some time
+    /// in the future. Regardless whether storing costs or the safety margin have changed
+    /// in the meantime (due to storage migration) a message can't be charged more than
+    /// deposited upfront.
+    ///
+    /// `id` - parameter convertible to the respective gas node id;
+    /// `hold_interval` - determines the time interval to charge rent for;
+    /// `cost` - rental cost per time unit (block);
+    /// `max_charge` - if set, contains the deposit locked at the time of starting storage rental.
     pub(crate) fn charge_for_hold(
```

Comment:
Seems we were discussing here ability to unlock funds within the fn: we can depend cost on lock type as well

Reply:
max charge would be queried in place tho

Reply:
Right. Can pull it inside the function. That would require default locking of the entire available gas for `Cut` nodes (which is currently not done yet). Anyway, we've discussed it already.